persistence-api/src/main/java/io/stargate/db/datastore/DataStoreFactory.java

```diff
@@ -35,6 +35,9 @@
    */
   DataStore create(@Nullable String userName, @Nonnull DataStoreOptions options);
 
+  DataStore create(
+      @Nullable String userName, boolean isFromExternalAuth, @Nonnull DataStoreOptions options);
```

Comment:
Why not use `AuthenticationSubject` as the parameter? It looks like we need most of its properties anyway?

Reply:
Trying to avoid a circular dependency between authnz and persistence-api

Reply:
ah, makes sense